evalml/data_checks/id_columns_data_check.py

```diff
@@ -175,9 +175,27 @@ def validate(self, X, y=None):
         ]  # columns whose name is "id"
         id_cols = {col: 0.95 for col in cols_named_id}
 
-        X = X.ww.select(include=["Integer", "Categorical"])
+        X_double = X.ww.select(include=["Double"])
 
-        check_all_unique = X.nunique() == len(X)
+        # Temporary solution for baton logical types mapping integers to doubles in woodwork logical types.
+        # Will be removed when resolved.
+        check_all_unique = X_double.nunique() == len(X_double)
+        cols_with_all_unique = check_all_unique[
+            check_all_unique
+        ].index.tolist()  # columns whose values are all unique and doubles
+        cols_with_all_unique_integers = [
+            col for col in cols_with_all_unique if all(X_double[col].mod(1).eq(0))
```

Comment:
I wonder if we can use
```python
col for col in cols_with_all_unique if all(X_double[col].is_integer())
```
to capture this logic

Reply:
This doesn't work unfortunately because `.is_integer` doesn't work for series types